pkg/server/server.go

```diff
@@ -361,46 +361,30 @@ func (s *ProxyServer) Proxy(stream client.ProxyService_ProxyServer) error {
 	userAgent := md.Get(header.UserAgent)
 	klog.V(2).InfoS("proxy request from client", "userAgent", userAgent)
 
-	recvCh := make(chan *client.Packet, 10)
```

Comment:
we're losing this 10 packet buffer. Having it could have possibly helped avoid grpc-layer lock-up issues between proxy-server and proxy-agent, similar to what we saw between KAS and proxy-server at #167.

Mentioning for awareness, not necessarily as a blocker.